packages/adblocker/src/engine/engine.ts

```diff
@@ -697,23 +766,46 @@ export default class FilterEngine extends EventEmitter<
     url,
     hostname,
     domain,
+
+    reference,
   }: {
     url: string;
     hostname: string;
     domain: string | null | undefined;
+
+    reference?: any | undefined;
   }): HTMLSelector[] {
     const htmlSelectors: HTMLSelector[] = [];
 
     if (this.config.enableHtmlFiltering === false || this.config.loadCosmeticFilters === false) {
       return htmlSelectors;
     }
 
-    const rules = this.cosmetics.getHtmlRules({
+    const { rules, matches, exceptions } = this.cosmetics.getHtmlRules({
       domain: domain || '',
       hostname,
       isFilterExcluded: this.isFilterExcluded.bind(this),
     });
 
+    const context: HtmlFilterMatchingContext = {
+      url,
+      hostname,
+      domain,
+
+      isFilterExcluded: this.isFilterExcluded.bind(this),
```

Comment:
this should definitely not be a part of the context, but I get the point pass all arguments, maybe this needs to be done differently

Reply:
Passing `isFilterExcluded` doesn't cause any load on adblocker library. Also, using the function from other places cannot interfere the internal process. We should decide this by focusing on the actual needs.

Reply:
it provides no information, but is leaking the internals.